exporter/opentelemetry-exporter-otlp-proto-grpc/src/opentelemetry/exporter/otlp/proto/grpc/exporter.py

```diff
@@ -321,15 +321,17 @@ def _export(self, data: TypingSequence[SDKDataT]) -> ExportResultT:
                         )
 
                     logger.warning(
-                        "Transient error %s encountered while exporting span batch, retrying in %ss.",
+                        "Transient error %s encountered while exporting %s, retrying in %ss.",
                         error.code(),
+                        data.__class__.__name__,
```

Comment:
How is this passing since data is a sequence? 
```python
data: TypingSequence[SDKDataT
````

Reply:
`data` may be a sequence but any sequence it is still an object like everything else, is an instance of a class and that class has a name.

We should update the type here, though, when metrics are exported `_export` receives [`MetricsData`](https://github.com/open-telemetry/opentelemetry-python/blob/main/exporter/opentelemetry-exporter-otlp-proto-grpc/src/opentelemetry/exporter/otlp/proto/grpc/metric_exporter/__init__.py#L200), when spans are exported, `_export` receives [`Sequence[ReadableSpan]`](https://github.com/open-telemetry/opentelemetry-python/blob/main/exporter/opentelemetry-exporter-otlp-proto-grpc/src/opentelemetry/exporter/otlp/proto/grpc/trace_exporter/__init__.py#L290).

Reply:
Ah thanks for the explanation. But it won't be helpful to say `Failed to export list ...`? I would suggest just say telemetry batch or something instead?

Reply:
Well, I think it is better if we can be more specific and tell the user at least what kind of thing is failing.

Reply:
The user will see a message that says a `list` could not be exported. I understand that is not very helpful but the handling of the failed export message is correct. The problem is not in this PR, but in the way we export traces, using a sequence of `ReadableSpan` instead of using an object that does follow closely the trace proto as we do for metrics (in case of a metrics export failure the user would get a useful warning that says `MetricsData` could not be exported).

Can we refactor the exported trace object to match the proto as we did for metrics? :thinking:

Reply:
That would probably be a breaking change right? Unless if we add backwards compatibility for exporters to accept both sequence of `ReadableSpan` and the proto definition.

To address the current issue, can we just do a check whether it's a `sequence` or not? And if it is, use the actual type of the object in the sequence to produce a meaningful error message?

Reply:
Ok, added fixed messages for traces and metrics.